lightningd/lightningd.c

```diff
@@ -182,6 +199,21 @@ static const char *find_my_path(const tal_t *ctx, const char *argv0)
 	tal_free(tmpctx);
 	return path_dirname(ctx, take(me));
 }
+static const char *find_my_pkglibexec_path(const tal_t *ctx,
+					   const char *my_path TAKES)
+{
+	const char *pkglibexecdir;
+	pkglibexecdir = path_join(ctx, my_path, BINTOPKGLIBEXECDIR);
+	return path_simplify(ctx, take(pkglibexecdir));
+}
+/* Determine the correct daemon dir. */
+static const char *find_daemon_dir(const tal_t *ctx, const char *argv0)
+{
+	const char *my_path = find_my_path(ctx, argv0);
+	if (has_all_daemons(my_path))
+		return my_path;
+	return find_my_pkglibexec_path(ctx, take(my_path));
```

Comment:
It seems weird that we dance around to calculate the bin-to-libexec mapping, then all we use it for is the get the libexec dir back?  Why not just return LIBEXECDIR here?  (With Makefile CPPFLAGS modified appropriately)?

Reply:
This makes searching for subdaemons work even if DESTDIR is non empty. 

Notice that we use this feature in `make installcheck`. We install with `DESTDIR=installcheck` and try out `installcheck$(bindir)/lightningd --test-daemons-only`. If we gave the LIBEXECDIR directly this check would not work as lightningd would use the absolute LIBEXECDIR. With the relative path weirdness this works even when executing lightningd in a DESTDIR.

We need installcheck since lightningd is still in flux. If we add a new subdaemon but forget to add it to install, we need something to catch it.